Fix: run_jobs return always an error !!

hregis opened the pull request.

hregis committed:
Fix: run_jobs return always an error !!

eldy commented on htdocs/cron/class/cronjob.class.php:
```diff
@@ -1021,7 +1021,7 @@ function run_jobs($userlogin)
 			}
 
-			if ($result === false || $result != 0)
+			if ($result === false || $result < 0)
```
Because some systems/os are not able to manage negative numbers for return on command line, the rule is that for cron method, 0 means ok, other means ko. To != 0 looks good to me.

hregis replied:
@eldy 
yes but if my $result = "true", with this test, $result not equal "false" OR not equal 0 !
maybe the best test is : 

if ($result === false && $result != 0)

you see ?

hregis replied:
@eldy 
no !

if ($result !== true || $result != 0)

hregis committed:
Fix: change the condition

hregis commented:
@eldy 
i change this PR with : 

if ($result !== true || $result != 0)

eldy commented:
Si $result = 0;
print ($result !== true)  renvoi vrai.

Donc cela ira en erreur alors que cela doit pas.
Je pige donc pas ta modif. Les jobs fournis par défaut avec dolibarr fonctionne bien en l'état. Tu as quoi dans le var_dump($result) avant ce test ?

hregis commented:
@eldy 
arrrf oui, tu as raison...
mon $result est "true", et si on laisse comme c'est j'ai une erreur, il faudrait trouver un équilibre pour un $result à true ou à 0 : 

$result = true;
if ($result === false || $result !=0)
	echo 'error';
else 
	echo 'OK';

hregis commented:
@eldy 
ça ça fonctionne pour true et 0

$result = true;
if ($result === false || (! is_bool($result) && $result != 0))
	echo 'error';
else 
	echo 'OK';

hregis committed:
Update cronjob.class.php

is better condition for use true or 0

if ($result === false || (! is_bool($result) && $result != 0))
